plugins/flow-visibility/clickhouse-monitor/main.go

```diff
@@ -0,0 +1,275 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"database/sql"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/ClickHouse/clickhouse-go"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+)
+
+const (
+	// The storage percentage at which the monitor starts to delete old records. By default, if the storage usage is larger than 50%, it starts to delete the old records.
+	threshold = 0.5
+	// The percentage of records in Clickhouse will be deleted when the storage is above threshold.
+	deletePercentage = 0.5
+	// The monitor stops for 3 intervals after a deletion to wait for the Clickhouse MergeTree Engine to release memory.
+	skipRoundsNum = 3
+	// Connection to Clickhouse timeout if if fails for 1 minute.
+	connTimeout = time.Minute
+	// Retry connection to Clickhouse every 5 seconds if it fails.
+	connRetryInterval = 5 * time.Second
+	// Query to Clickhouse timeout if if fails for 10 seconds.
+	queryTimeout = 10 * time.Second
+	// Retry query to Clickhouse every second if it fails.
+	queryRetryInterval = 1 * time.Second
+	// Time format for timeInserted
+	timeFormat = "2006-01-02 15:04:05"
+)
+
+var (
+	// The name of the table to store the flow records
+	tableName = os.Getenv("TABLE_NAME")
+	// The names of the materialized views
+	mvNames = strings.Split(os.Getenv("MV_NAMES"), " ")
+	// The namespace of the Clickhouse server
+	namespace = os.Getenv("NAMESPACE")
+	// The clickhouse monitor label
+	monitorLabel = os.Getenv("MONITOR_LABEL")
+)
```

Comment:
it would be good to fail early in `main()` if one of the required environment variables is missing. What do you think?